.github/workflows/test_pr_and_main.yml

```diff
@@ -29,7 +29,7 @@ env:
   PYTHON_NUMPY_PKGS: >
       numpy scipy pyodbc pandas matplotlib seaborn numdifftools 
       parameterized casadi
-  CACHE_VER: v210415.0
+  CACHE_VER: v4
```

Comment:
We are adopting a newer "date-based" identifier.  When you re-sync with main, either preserve the `v210415.0` or update it to today's date (`vYYMMDD.n` format)